File: app/src/main/java/com/mlsdev/mlsdevstore/presentaion/account/EditPersonalInfoActivity.java
package com.mlsdev.mlsdevstore.presentaion.account;

import androidx.lifecycle.ViewModelProviders;
import android.content.Context;
import android.content.Intent;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.Observable;
import android.os.Bundle;

import com.mlsdev.mlsdevstore.R;
import com.mlsdev.mlsdevstore.databinding.ActivityEditPersonalInfoBinding;
import com.mlsdev.mlsdevstore.presentaion.BaseActivity;
import com.mlsdev.mlsdevstore.presentaion.utils.CustomObservableBoolean;

public class EditPersonalInfoActivity extends BaseActivity {
    private ActivityEditPersonalInfoBinding binding;
    private EditAccountViewModel viewModel;

    public static void launch(Context context) {
        context.startActivity(new Intent(context, EditPersonalInfoActivity.class));
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_edit_personal_info);
        initViewModel();
        initToolbar(binding.toolbar);
        displayBackArrow(true);
    }

    @Override
    protected void onDestroy() {
        getLifecycle().removeObserver(viewModel);
        viewModel.accountUpdated.removeOnPropertyChangedCallback(accountUpdatedCallback);
        super.onDestroy();
    }

    private void initViewModel() {
        viewModel = ViewModelProviders.of(this, viewModelFactory).get(EditAccountViewModel.class);
        viewModel.accountUpdated.addOnPropertyChangedCallback(accountUpdatedCallback);
        binding.setViewModel(viewModel);
        getLifecycle().addObserver(viewModel);
    }

    private Observable.OnPropertyChangedCallback accountUpdatedCallback = new Observable.OnPropertyChangedCallback() {
        @Override
        public void onPropertyChanged(Observable observable, int i) {
            if (observable instanceof CustomObservableBoolean && ((CustomObservableBoolean) observable).get())
                finish();
        }
    };
}

File: app/src/main/java/com/mlsdev/mlsdevstore/presentaion/account/EditPersonalInfoActivity.kt
package com.mlsdev.mlsdevstore.presentaion.account

import androidx.lifecycle.ViewModelProviders
import android.content.Context
import android.content.Intent
import androidx.databinding.DataBindingUtil
import androidx.databinding.Observable
import android.os.Bundle

import com.mlsdev.mlsdevstore.R
import com.mlsdev.mlsdevstore.databinding.ActivityEditPersonalInfoBinding
import com.mlsdev.mlsdevstore.presentaion.BaseActivity
import com.mlsdev.mlsdevstore.presentaion.utils.CustomObservableBoolean

class EditPersonalInfoActivity : BaseActivity() {
    lateinit var binding: ActivityEditPersonalInfoBinding
    lateinit var viewModel: EditAccountViewModel

    private val accountUpdatedCallback = object : Observable.OnPropertyChangedCallback() {
        override fun onPropertyChanged(observable: Observable, i: Int) {
            if (observable is CustomObservableBoolean && observable.get())
                finish()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_edit_personal_info)
        initViewModel()
        initToolbar(binding.toolbar)
        displayBackArrow(true)
    }

    override fun onDestroy() {
        lifecycle.removeObserver(viewModel)
        viewModel.accountUpdated.removeOnPropertyChangedCallback(accountUpdatedCallback)
        super.onDestroy()
    }

    private fun initViewModel() {
        viewModel = ViewModelProviders.of(this, viewModelFactory).get(EditAccountViewModel::class.java)
        viewModel.accountUpdated.addOnPropertyChangedCallback(accountUpdatedCallback)
        binding.viewModel = viewModel
        lifecycle.addObserver(viewModel)
    }
}
